test/NuGet.Clients.Tests/NuGet.CommandLine.Test/CommandRunner.cs

```diff
@@ -29,6 +26,7 @@ public class CommandRunner
                 RedirectStandardError = true,
                 RedirectStandardInput = inputAction != null
             };
+            psi.EnvironmentVariables["NuGetTestModeEnabled"] = "True";
```

Comment:
This still keeps all the unit tests that do not use the command line, not setting the environment variable